econml/dml.py

```diff
@@ -764,7 +914,69 @@ def fit(self, Y, T, X=None, W=None, *, sample_weight=None, sample_var=None, grou
                                "We recommend using the LinearDML estimator for this low-dimensional setting.")
         return super().fit(Y, T, X=X, W=W,
                            sample_weight=sample_weight, sample_var=None, groups=groups,
-                           inference=inference)
+                           cache_values=cache_values, monte_carlo_iterations=None, inference=inference)
+
+    @DML.model_final.setter
```

Comment:
I feel that for all these setters, there should be a logic within a __setattr__, as opposed to a separate method for each one, just so that we reduce lines of code. But this is secondary and I'm ok with doing it post release.